modules/core/src/main/scala/aecor/es/DomainEvent.scala

```diff
@@ -0,0 +1,6 @@
+package aecor.es
+
+/**
+  * Market trait
+  */
+trait DomainEvent extends Product with Serializable
```

Comment:
Why do we need this?

Reply:
Part of this exercise is to document the concepts for code comprehension purposes.
Code is very rich in design decisions and it would beneficial to relate it to other implemenations of `event sourcing` for better understanding.

In particular, `DomainEvent` is similar to https://github.com/debasishg/frdomain/blob/48664c35b2e583aeacfcacd5d740687afb2ed1de/src/main/scala/frdomain/ch8/cqrs/lib/es.scala#L20

Both this and `DomainState` are marker or tagging trait (https://en.wikipedia.org/wiki/Marker_interface_pattern). Please allow such things, again, for code comprehension purposes (can be used to relate across modules and applications).

Reply:
@SemanticBeeng I agree with @vpavkin here. There is no need for such types. Just add `Event` suffix to your real event ADT. Marker traits are used in weakly typed and/or runtime reflection based libraries (which `Aecor` is not), e.g. Akka uses it for serializers that accept `Any`.